### PR TITLE
Eb sprint 1 functionality

### PR DESCRIPTION
**Pull request by ericberglund117**
## Description
- Created sad path testing for pantry with no ingredients
- Created sad path testing to favoriteRecipes function
- Refactor saveRecipe function
- Created sad path testing to decideToCook if the recipe is undefined
- Created sad path to filter/search functions to return null if arguments are undefined
- Add proper syntax to files
- Add smaller data set to recipe-test.js
- Create calculateIngredientCost functionality to pass test

## Motivation and Context
- These changes were required for a more robust testing structure 
- The functions in the Recipe class also needed refactoring in order to successfully work
- Refactor the calculateIngredientCost to determine the total amount in cents that each recipe requires. This is based on the amount of the ingredient that is needed for the recipe times the estimatedCostInCents from the ingredientsData

## How Has This Been Tested?
- This has been tested through happy and sad path testing. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test/recipe-test.js [hunk 1]
```diff
@@ -21,15 +87,15 @@ describe('Recipe', function() {
...
-    expect(recipe.name).to.eq('Loaded Chocolate Chip Pudding Cookie Cups');
+    expect(recipe.name).to.equal('Loaded Chocolate Chip Pudding Cookie Cups');
   });
 
   it('should initialize with an image', function() {
-    expect(recipe.image).to.eq('https://spoonacular.com/recipeImages/595736-556x370.jpg');
+    expect(recipe.image).to.equal('https://spoonacular.com/recipeImages/595736-556x370.jpg');
   });
 
   it('should initialize with an array of ingredients', function() {
```
Comment (Kristiannmiller) [inline test/recipe-test.js:101]: Looks like we might have overlooked a test. Let's get this test written with the next branch. Should be a quick fix!

File: test/user-test.js [hunk 2]
```diff
@@ -77,17 +77,26 @@ describe('User', function() {
...
   it('should initialize with a name', function() {
-    expect(user1.name).to.eq('Saige O\'Kon');
+    expect(user1.name).to.equal('Saige O\'Kon');
   });
 
   it('should initialize with a pantry', function() {
     expect(user1.pantry[0].ingredient).to.eq(11477);
   });
 
+  it('should return 0 if there are no ingredients in the panty', function() {
```
Comment (Kristiannmiller) [inline test/user-test.js:91]: Great sad path test! Michael Scarn would be proud.

File: test/user-test.js [hunk 3]
```diff
@@ -101,18 +110,42 @@ describe('User', function() {
     expect(user1.favoriteRecipes[0].name).to.equal('Chicken Parm');
   });
 
+  it('should return an alert message if a recipe is undefined', function() {
```
Comment (Kristiannmiller) [inline test/user-test.js:113]: Did we get this to send an alert, or return the result? We were having trouble with how to test an alert if I remember right. Maybe the should statement needs changed?

**Comment by Kristiannmiller** [comment]
Looks great! If @chadrick-d-dev give the thumbs up too let's get this branch merged!

File: test/user-test.js (hunk 3 above)
Comment (ericberglund117) [reply test/user-test.js:113]: That's a good point. I forgot that we had issues with the alert function. We might need to change the test or fix our function

File: test/user-test.js (hunk 2 above)
Comment (chadrick-d-dev) [reply test/user-test.js:91]: Yes I concur, very good use of sad testing!

File: test/recipe-test.js (hunk 1 above)
Comment (chadrick-d-dev) [reply test/recipe-test.js:101]: Great catch Kristi!

**Comment by chadrick-d-dev** [comment]
Looks great guys, this is ready to merge!

File: test/user-test.js [hunk 4]
```diff
@@ -101,18 +110,42 @@ describe('User', function() {
...
+    user1.saveRecipe(recipe1)
+    expect(user1.filterRecipes(undefined)).to.equal(null)
+  })
+
   it('should be able to search recipes by name', function() {
     user1.saveRecipe(recipe);
     expect(user1.searchForRecipe('Chicken Parm')).to.deep.equal([recipe]);
   });
+
+  it('should return null if the name is undefined in the searchForRecipe function', function() {
```
Comment (chadrick-d-dev) [inline test/user-test.js:146]: Another great example of sad testing! Approve approve!

**Review by chadrick-d-dev — APPROVED** [review]
All and all, I love all the contributions, and like Kristi said, let's work out a few details.  Great work team!

**Pull request merged**